lib/ohai/plugins/dragonflybsd/platform.rb

```diff
@@ -0,0 +1,29 @@
+#
+# Author:: Bryan McLellan (btm@loftninjas.org)
+# Copyright:: Copyright (c) 2009 Bryan McLellan
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+Ohai.plugin(:Platform) do
+  provides "platform", "platform_version", "platform_family"
+
+  collect_data(:dragonflybsd) do
+    so = shell_out("uname -s")
+    platform so.stdout.split($/)[0].downcase
+    so = shell_out("uname -r")
+    platform_version so.stdout.split($/)[0]
+    platform_family "dragonflybsd"
```

Comment:
They all seem to have completely different package managers and init systems/service management.  I just can't point to a lot of code like `%w{openbsd netbsd freebsd}.include?(node[:os])` where it'd actually be useful to replace that with a platform_family.